### DISCUSSION TITLE
Logging the SQL only if there's a problem?

### DISCUSSION THREAD

**Discussion by ksze** [sequelize/sequelize#13778] [Help & Questions]
Is there a way to configure sequelize v6 such that it logs the SQL only if there's a problem - e.g. missing bindings, or UniqueConstraintError, or ForeignKeyConstraintError?

**Comment by fzn0x** [comment]
Sequelize use the debug module, you can setup cross-env and use `DEBUG=sequelize:sql*` before starting your app.

**Comment by ephys** [comment]
Correct me if I'm wrong but that's going to log every SQL queries, not just those that had a problem.

However, when an error is thrown, the query is typically included in the Error object.

**Reply by PythonCoderAS** [reply]
Yes, from my experience the error object usually includes a copy of the query.

**Reply by fzn0x** [reply]
Yes, you are not wrong, it's going to log every SQL queries, but I find that approach to be good too since it will stop on throwed error.

Maybe you can be more specific or objective about the query that included in the Error object.
